scripts/policy/frameworks/management/persistence.zeek

```diff
@@ -0,0 +1,52 @@
+##! Common adjustments for any kind of Zeek node when we run the Management
+##! framework.
+
+@load base/misc/installation
+@load base/utils/paths
+
+@load ./config
+
+# For testing, keep persistent state local to the current working directory,
+# and disable log rotation.
+@if ( getenv("ZEEK_MANAGEMENT_TESTING") != "" )
+
+redef Management::spool_dir = ".";
+redef Management::state_dir = ".";
+redef Log::default_rotation_interval = 0 secs;
+
+@else
+
+# For any kind of Zeek process we steer rotated logs awaiting archival into a
+# queue directory in the spool.
+redef Log::default_rotation_dir = build_path(Management::get_spool_dir(), "log-queue");
+
+# Additional default storage locations:
+redef Broker::table_store_db_directory = build_path(Management::get_state_dir(), "broker");
+@if ( getenv("CLUSTER_NODE") != "" )
+redef Cluster::default_store_dir = build_path(Management::get_state_dir(), "cluster");
+@endif
+
+@if ( getenv("ZEEK_MANAGEMENT_NODE") != "" )
+
+# Management agents and controllers don't have loggers, nor their configuration,
+# so establish a similar one here:
+
+function archiver_rotation_format_func(ri: Log::RotationFmtInfo): Log::RotationPath
+	{
+	local open_str = strftime(Log::default_rotation_date_format, ri$open);
+	local close_str = strftime(Log::default_rotation_date_format, ri$close);
+	local base = fmt("%s__%s__%s__", ri$path, open_str, close_str);
+	local rval = Log::RotationPath($file_basename=base);
+	return rval;
+	}
+
+redef Log::default_rotation_interval = 1 hrs;
+redef Log::enable_local_logging = T;
+redef Log::enable_remote_logging = F;
```

Comment:
I'm wondering if they should be logging remotely as well (or even only? not sure).

Reply:
I think "as well" is a good call, yeah. There's currently nothing that subscribes to logs from agent/controller, so I figured I'd keep it off, but that might change soon when we expand the client's ability to receive logs. For now I don't think we can get by without local logging for these. I'll push an update.

Reply:
Would be neat if these logs came out through the logger as well, though that's a chicken-and-egg I guess ...